From 34993875099d41dfbedb7d0f31e6e788daa41838 Mon Sep 17 00:00:00 2001
From: Huiquan Zhong <huiquan.zhong@intel.com>
Date: Sat, 18 Oct 2014 00:44:10 +0800
Subject: [PATCH 1/2] dma: dw: enable CherryTrail LPDMA

Add CherryTrail's LPDMA ACPI ID(80862286, 808622C0) to the
mapping table.

Change-Id: I58cfd9cd404e612c3df745c1dad8c21cef369416
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-2493
Signed-off-by: Huiquan Zhong <huiquan.zhong@intel.com>
---
 drivers/acpi/acpi_lpss.c  |    4 ++--
 drivers/dma/dw/platform.c |    2 ++
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/acpi/acpi_lpss.c b/drivers/acpi/acpi_lpss.c
index 1b3a6bbeb33f..5ed8e7290dc2 100644
--- a/drivers/acpi/acpi_lpss.c
+++ b/drivers/acpi/acpi_lpss.c
@@ -195,8 +195,8 @@ static const struct acpi_device_id acpi_lpss_device_ids[] = {
 	/* Cherrytrail LPSS devices */
 	{ "808622C1", (unsigned long)&cht_i2c_dev_desc },
 	{ "8086228A", (unsigned long)&byt_uart_dev_desc },
-	{ "80862286", }, /* CHT DMA1 */
-	{ "808622C0", }, /* CHT DMA2 */
+	{ "80862286", (unsigned long)&lpss_dma_desc },
+	{ "808622C0", (unsigned long)&lpss_dma_desc },
 	{ "8086228E", (unsigned long)&byt_spi_dev_desc },
 	{ "80862288", }, /* CHT PWM */
 
diff --git a/drivers/dma/dw/platform.c b/drivers/dma/dw/platform.c
index c872392c71e0..21c9da1108f8 100644
--- a/drivers/dma/dw/platform.c
+++ b/drivers/dma/dw/platform.c
@@ -252,6 +252,8 @@ MODULE_DEVICE_TABLE(of, dw_dma_of_id_table);
 #ifdef CONFIG_ACPI
 static const struct acpi_device_id dw_dma_acpi_id_table[] = {
 	{ "INTL9C60", 0 },
+	{ "80862286", 0 },
+	{ "808622C0", 0 },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, dw_dma_acpi_id_table);
-- 
1.7.9.5

